src/main/java/org/opensearch/knn/quantization/models/quantizationState/OneBitScalarQuantizationState.java

```diff
@@ -0,0 +1,181 @@
+     * It includes versioning information to ensure compatibility between different versions of the serialized object.
+     *
+     * <p>The version information is read first, and then it is set using the {@link VersionContext#setVersion(int)} method.
+     * This makes the version information available to all classes involved in the deserialization process within the current thread context.</p>
+     *
+     * <p>Classes that are part of the deserialization process can retrieve the version information using the
+     * {@link VersionContext#getVersion()} method and implement version-specific logic accordingly.</p>
+     *
+     * <pre>
+     * {@code
+     * // Example usage in the readExternal method:
+     * int version = in.readInt(); // Read the version
+     * VersionContext.setVersion(version);
+     * quantizationParams = new ScalarQuantizationParams();
+     * quantizationParams.readExternal(in); // Use readExternal of SQParams
+     * int length = in.readInt();
+     * meanThresholds = new float[length];
+     * for (int i = 0; i < length; i++) {
+     *     meanThresholds[i] = in.readFloat();
+     * }
+     * }
+     * </pre>
+     *
+     * @param in the ObjectInput to read the object from.
+     * @throws IOException if an I/O error occurs during deserialization.
+     * @throws ClassNotFoundException if the class of the serialized object cannot be found.
+     */
+    @Override
+    public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+        int version = in.readInt(); // Read the version
+        VersionContext.setVersion(version);
+        quantizationParams = new ScalarQuantizationParams();
+        quantizationParams.readExternal(in); // Use readExternal of SQParams
+        int length = in.readInt();
+        meanThresholds = new float[length];
+        for (int i = 0; i < length; i++) {
+            meanThresholds[i] = in.readFloat();
+        }
+        VersionContext.clear(); // Clear the version after use
+    }
+
+    /**
+     * Serializes the current state of this OneBitScalarQuantizationState object into a byte array.
+     * This method uses the QuantizationStateSerializer to handle the serialization process.
+     *
+     * <p>The serialized byte array includes all necessary state information, such as the mean thresholds
+     * and quantization parameters, ensuring that the object can be fully reconstructed from the byte array.</p>
+     *
+     * <pre>
+     * {@code
+     * OneBitScalarQuantizationState state = new OneBitScalarQuantizationState(params, meanThresholds);
+     * byte[] serializedState = state.toByteArray();
+     * }
+     * </pre>
+     *
+     * @return a byte array representing the serialized state of this object.
+     * @throws IOException if an I/O error occurs during serialization.
+     */
+    @Override
+    public byte[] toByteArray() throws IOException {
```

Comment:
Do we need this if we implement Writeable interface?

Reply:
I commented on top for this.